ansible/roles/debops.grafana/defaults/main.yml

```diff
@@ -0,0 +1,193 @@
+# ================================
+
+# .. contents:: Sections
+#    :local:
+#
+# .. include:: ../../../../includes/global.rst
+
+
+# Domain name configuration [[[
+# -----------------------------
+
+# .. envvar:: grafana__fqdn [[[
+#
+# String or list of the Fully Qualified domain names on which the Grafana
+# application will be available, used by the webserver. The first item will be
+# used as the e-mail sender domain name.
+grafana__fqdn: 'grafana.{{ grafana__domain }}'
+
+                                                                   # ]]]
+# .. envvar:: grafana__domain [[[
+#
+# The DNS domain used by other variables in the ``debops.grafana`` role.
+grafana__domain: '{{ ansible_local.core.domain
+                    if (ansible_local|d() and ansible_local.core|d() and
+                        ansible_local.core.domain|d())
+                    else ansible_domain }}'
+
+                                                                   # ]]]
+                                                                   # ]]]
+# Grafana configuration [[[
+# --------------------------------
+
+# .. envvar:: grafana__nginx_enabled: True [[[
+#
+# Enable or disable Nginx dependencies.
+grafana__nginx_enabled: True
+
+                                                                   # ]]]
+# .. envvar:: grafana__upstream_apt_id [[[
+#
+# The Grafana fingerprint.
+grafana__upstream_apt_id: '4E40 DDF6 D76E 284A 4A67  80E4 8C8C 34C5 2409 8CB6'
+
+                                                                   # ]]]
+# .. envvar:: grafana__upstream_apt_repository [[[
+#
+# The APT repository URI of the upstream Grafana repository.
+grafana__upstream_apt_repository: 'deb https://packages.grafana.com/oss/deb stable main'
+
+                                                                   # ]]]
+                                                                   # ]]]
+# Installation, APT packages [[[
+# ------------------------------
+
+# .. envvar:: grafana__base_packages [[[
+#
+# List of APT packages which are required by the Grafana role.
+grafana__base_packages: [ 'grafana' ]
+
+                                                                   # ]]]
+# .. envvar:: grafana__packages [[[
+#
+# List of APT packages to install on all hosts in Grafana inventory.
+grafana__packages: [ 'software-properties-common', 'apt-transport-https' ]
+
+                                                                   # ]]]
+# .. envvar:: grafana__deploy_type [[[
+#
+# Select how Grafana should be installed by the role:
+#
+# - ``system``: install Grafana using current OS packages, possibly from the
+#               backports repository.
+#
+# - ``upstream``: install Grafana from the upstream APT repository on
+#                 Launchpad (PPA, usable on Debian as well).
+#
+# - ``bootstrap``: install APT packages required for Grafana and build a local
+#                  ``.deb`` package using the upstream GitHub repository.
+#
+grafana__deploy_type: "upstream"
```

Comment:
Looks like you based this variable on `debops.ansible` role. :-)

Currently Grafana is not present in any Debian repositories. Ubuntu has it in the `xenial` archive, but not anywhere else. It seems that the OS distributions don't plan to package Grafana anytime soon, therefore I would remove the `system` option from here.

The `bootstrap` method is used in `debops.ansible` to allow installation of a newer version for testing and development purposes - I would mark it as an exception to the rule since DebOps is based on Ansible. In the Grafana case though, I wouldn't want to build it from source, since upstream provides APT repository which makes upgrades of new releases painless. So, I would drop the `bootstrap` installation type as well.

This leaves the `upstream` as the only type and makes the variable redundant. It can be safely removed to make conditions in tasks easier.

Reply:
also deleted